src/yarp.c

```diff
@@ -5201,7 +5201,7 @@ parse_expression_prefix(yp_parser_t *parser) {
         yp_node_t *else_node;
 
         if (!match_type_p(parser, YP_TOKEN_KEYWORD_END)) {
-          else_node = yp_node_else_node_create(parser, &else_keyword, parse_statements(parser, YP_CONTEXT_CASE_WHEN), &parser->previous);
+          else_node = yp_node_else_node_create(parser, &else_keyword, parse_statements(parser, YP_CONTEXT_CASE_WHEN), &parser->current);
```

Comment:
This is definitely the right change, but we shouldn't be using `YP_CONTEXT_CASE_WHEN` here, we should be passing `YP_CONTEXT_ELSE`. Can you change that?